LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs

```diff
@@ -102,9 +121,11 @@ public LoRaDevice(string devAddr, string devEUI, ILoRaDeviceClient loRaDeviceCli
             this.IsABPRelaxedFrameCounter = true;
             this.PreferredWindow = 1;
             this.hasFrameCountChanges = false;
-            this.fcntLock = new object();
+            this.fcntSync = new object();
             this.confirmationResubmitCount = 0;
+            this.queueSync = new object();
             this.queuedRequests = new Queue<LoRaRequest>();
+            this.classType = LoRaDeviceClassType.A;
```

Comment:
All of the initializers that are not passed into the ctor could be directly assigned when you declare the field. The advantage I see with that is that if you ever add another contstructor overload, you have the fields set to the correct defaults. 

I think I made that comment before, and it's probably personal preference, but the defaults for the various types, would not need to be re-initialized (it's additional code that's running):

bool is by default false (does not need to be initialized)
int is by default 0
classType is by default type.A

Reply:
I see the advantage of setting the value together with the property declaration, what I dislike is that during a debugging session stepping into the ctor will go through many lines. But, this is probably better than introducing bugs by adding constructors. Another way to address this issues is by having ctors calling other ctors.

Regarding initialising members with default values, specifically enums, the advantage I see is that the code is resilient to changes to the enum definition (i.e. defining as first enum None or NotSet could break the code).